common/network-types/src/session/frame.rs

```diff
@@ -158,6 +158,15 @@ impl Segment {
     /// Size of the segment header.
     pub const HEADER_SIZE: usize = mem::size_of::<FrameId>() + 2 * mem::size_of::<SeqNum>();
 
+    /// The minimum size of a segment: [`Segment::HEADER_SIZE`] + 1 byte of data.
+    pub const MINIMUM_SIZE: usize = Self::HEADER_SIZE + 1;
+
+    /// The maximum size of a segment: [`Segment::HEADER_SIZE`] + data, regardless
+    /// any MTU.
+    /// This number is currently set, so that segment length is expressible
+    /// with just 10-bits.
+    pub const MAXIMUM_SIZE: usize = 1500;
```

Comment:
10 bits would be `1024`.

Reply:
Check your math: 1500 = 1024 + 256 + 128 + 64 + 16 + 8 + 4 

Still fits within 10 bits ;)

Reply:
Weekends. 🍻